### PR TITLE
[AIRFLOW-4033] record stats of task duration

### PR DESCRIPTION
**Pull request by milton0825**
Make sure you have checked _all_ steps below.

### Jira

- [X] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-4033
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.
  - In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).

### Description

- [X] Here are some details about my PR, including screenshots of any UI changes:
Sending stats for task execution duration

### Tests

- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### Code Quality

- [X] Passes `flake8`

### CODE REVIEW

File: airflow/models/__init__.py [hunk 1]
```diff
@@ -1400,6 +1403,14 @@ def signal_handler(signum, frame):
 
                 task_copy.post_execute(context=context, result=result)
 
+                end_time = time.time()
+                duration = end_time - start_time
+                Stats.timing(
+                    '{dag_id}.{task_id}.duration'.format(
```
Comment (mik-laj) [inline airflow/models/__init__.py]: Can you update the documentation in file `docs/metrics.rst`?   I would be happy if you would also add other missing keys. 
Reference: 
My WIP PR that add missing keys - 
https://github.com/PolideaInternal/airflow/pull/79/commits/bed9e3bd26a242abb5a0ef1bf35e7d44ab1426f1

Comment (milton0825) [reply airflow/models/__init__.py]: I have updated the docs!

Comment (milton0825) [reply airflow/models/__init__.py]: @mik-laj you want me to take over your PR?

**Comment by milton0825** [comment]
@feng-tao @Fokko

File: airflow/models/__init__.py (hunk 1 above)
Comment (ashb) [inline airflow/models/__init__.py]: Do we need to worry about non-safe characters in the dag or task id, or maximum length?

File: docs/metrics.rst [hunk 2]
```diff
@@ -66,4 +66,5 @@ Timers
 Name                              Description
 ================================= =======================================
 dagrun.dependency-check.<dag_id>  Seconds taken to check DAG dependencies
+<dag_id>.<task_id>.duration       Seconds taken to finish a task
```
Comment (ashb) [inline docs/metrics.rst]: I feel this should be prefixed with something to make the metrics un-ambigious. `dag.`? (so `dag.mydag.my-task.duration = 300`)

File: airflow/models/__init__.py (hunk 1 above)
Comment (mik-laj) [reply airflow/models/__init__.py]: if you have time then I would be happy. However, I do not require it. I gave the link to show that other things could be corrected in the documentation. I do not care about finishing my PR.

File: docs/metrics.rst (hunk 2 above)
Comment (milton0825) [reply docs/metrics.rst]: yeah that make sense.

File: airflow/models/__init__.py (hunk 1 above)
Comment (milton0825) [reply airflow/models/__init__.py]: I think that worth another PR to fix it.

Comment (milton0825) [reply airflow/models/__init__.py]: https://github.com/apache/airflow/pull/4889

**Comment by milton0825** [comment]
PTAL @ashb

**Review by KevinYang21 — APPROVED** [review]
very useful, thanks

**Comment by Fokko** [comment]
I've restarted the failing tests.

**Comment by milton0825** [comment]
@Fokko thanks for restarting the failing tests. Looks like they passed.

**Review by feng-tao — APPROVED** [review]

**Pull request merged**